apps/emqx_resource/src/emqx_resource_buffer_worker.erl

```diff
@@ -244,15 +243,17 @@ code_change(_OldVsn, State, _Extra) ->
 
 %%==============================================================================
 -define(PICK(ID, KEY, EXPR),
```

Comment:
The macro is just called in two places. Maybe it is better to remove the macro and write out the code explicitly in these two places. This might increase code duplication but might also make the code easier to follow. We could also rewrite the macro to a function and pass the expression as a fun. Is the reason for using a macro here performance or anything else?

Reply:
Can't be performance since the resulting syntax tree is the same.

Reply:
Ok, I see. A function instead of a macro might still be an option but I think this is good enough.